UI/GovernmentWnd.cpp

```diff
@@ -1468,15 +1596,51 @@ void GovernmentWnd::Refresh() {
     m_main_panel->Refresh();
 }
 
+double GovernmentWnd::GetPolicyZoomFactor() {
+    switch (m_policy_size_buttons->CheckedButton()) {
+    case 0: return 1;
+    case 1: return 0.75;
+    case 2: return 0.5;
+    }
+
+    DebugLogger() << "GovernmentWnd::GetPolicyZoomFactor(): m_policy_buttons no button checked.";
+    return 1;
+}
+
+GG::Pt GovernmentWnd::GetPolicySlotSize() {
+    double zoom_factor = GetPolicyZoomFactor();
+    const GG::X slot_width = static_cast<GG::X>(SLOT_CONTROL_WIDTH * zoom_factor);
+    const GG::Y slot_height = static_cast<GG::Y>(SLOT_CONTROL_HEIGHT * zoom_factor);
+    return GG::Pt(slot_width, slot_height);
+}
+
+int GovernmentWnd::GetPolicyTextSize() {
+    double zoom_factor = GetPolicyZoomFactor();
+    const int text_pts = static_cast<int>(ClientUI::Pts() * zoom_factor);
+    return text_pts;
+}
+
 void GovernmentWnd::DoLayout() {
+    const GG::Y BUTTON_HEIGHT(ClientUI::Pts() * 2);
     constexpr GG::Pt palette_ul(GG::X0, GG::Y0);
     const GG::Pt palette_lr(palette_ul + GG::Pt(ClientWidth(), ClientHeight() / 2));
+    const int num_size_buttons = static_cast<int>(m_policy_size_buttons->NumButtons());
 
-    const GG::Pt main_ul(palette_ul + GG::Pt(GG::X0, ClientHeight() / 2));
+    const GG::Pt main_ul(palette_ul + GG::Pt(GG::X0, ClientHeight() / 2 + PAD));
     const GG::Pt main_lr(ClientWidth(), ClientHeight() - GG::Y(INNER_BORDER_ANGLE_OFFSET));
 
     m_main_panel->SizeMove(main_ul, main_lr);
     m_policy_palette->SizeMove(palette_ul, palette_lr);
+
+    const GG::Pt size_buttons_ul = main_ul + GG::Pt(GG::X(PAD), - BUTTON_HEIGHT/2);
+    const GG::Pt size_buttons_lr = main_ul + GG::Pt((GG::X(PAD) + POLICY_SIZE_BUTTON_WIDTH) * num_size_buttons, BUTTON_HEIGHT/2);
+
+    m_policy_size_buttons->SizeMove(size_buttons_ul, size_buttons_lr);
+    // m_policy_size_buttons->ExpandButtons(true);       // leads to crash
```

Comment:
Inexplicably, this leads to a crash - any idea why?

Reply:
Not sure. The error message for me in Visual Studio says heap corruption.

Stack looks like:
````
 	GiGi.dll!GG::Wnd::DetachChild(GG::Wnd * wnd) Line 622	C++
>	GiGi.dll!GG::Layout::Remove(GG::Wnd * wnd) Line 542	C++
 	GiGi.dll!GG::Layout::DetachChildCore(GG::Wnd * wnd) Line 734	C++
 	GiGi.dll!GG::Wnd::DetachChild(GG::Wnd * wnd) Line 622	C++
 	GiGi.dll!GG::Layout::ResizeLayout(unsigned int rows, unsigned int columns) Line 582	C++
 	GiGi.dll!GG::RadioButtonGroup::RemoveButton(GG::StateButton * button) Line 737	C++
 	GiGi.dll!GG::RadioButtonGroup::ExpandButtons(bool expand) Line 749	C++
 	FreeOrion.exe!GovernmentWnd::DoLayout() Line 1639	C++
 	FreeOrion.exe!GovernmentWnd::SizeMove(const GG::Pt & ul, const GG::Pt & lr) Line 1585	C++
 	[Inline Frame] FreeOrion.exe!CUIWnd::ValidatePosition() Line 237	C++
 	FreeOrion.exe!CUIWnd::CompleteConstruction() Line 172	C++
 	FreeOrion.exe!GovernmentWnd::CompleteConstruction() Line 1578	C++
 	FreeOrion.exe!GG::Wnd::Create<GovernmentWnd,std::string const &>(const std::string & <args_0>) Line 348	C++
 	FreeOrion.exe!MapWnd::CompleteConstruction() Line 1464	C++
 	FreeOrion.exe!GG::Wnd::Create<MapWnd>() Line 348	C++
````

Reply:
Found it: had to call RadioButtonGroup::ExpandButtons right after creating the RadioButtonGroup.